### PR TITLE
Add tool: Google Translate

### PR DESCRIPTION
**Pull request by svcvit**
# Description

Add tool: Google Translate

Xnip2024-07-10_21-30-13
Xnip2024-07-10_21-30-37

Xnip2024-07-10_21-30-46

Fixes 

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)
- [ ] Improvement, including but not limited to code refactoring, performance optimization, and UI/UX improvement
- [ ] Dependency upgrade

### CODE REVIEW

**Comment by hjlarry** [comment]
suggesion:
1. we should not use the `py-trans` pkg, it is an uncommon pkg, and here it only sent a request to google, you can sent it yourself
2. the `destination language` should be an option, users can't remember each language code
3. if the `pt_BR` is same as `en_US`, you can remove it

**Comment by svcvit** [comment]
Great advice, let me rewrite the code

**Comment by svcvit** [comment]
Rewrite the translation function, remove `pt_BR`, modify the type of `destination language`, change it to `select`.

File: api/core/tools/provider/builtin/google_translate/tools/translate.yaml [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+  - name: dest
+    type: select
+    required: true
+    label:
+      en_US: destination language
+      zh_Hans: 目标语言
+    human_description:
+      en_US: The destination language you want to translate.
+      zh_Hans: 你想翻译的目标语言
+    required: false
```
Comment (crazywoola) [inline api/core/tools/provider/builtin/google_translate/tools/translate.yaml]: Here is a duplicated `required` key.

**Review by crazywoola — DISMISSED** [review]

**Review by crazywoola — DISMISSED** [review]

**Review by crazywoola — APPROVED** [review]

**Pull request merged**